packages/flutter/lib/src/material/date_picker_theme.dart

```diff
@@ -0,0 +1,930 @@
+    this.headerForegroundColor,
+    this.headerHeadlineStyle,
+    this.headerHelpStyle,
+    this.weekdayStyle,
+    this.dayStyle,
+    this.dayForegroundColor,
+    this.dayBackgroundColor,
+    this.dayOverlayColor,
+    this.todayForegroundColor,
+    this.todayBackgroundColor,
+    this.todayBorder,
+    this.yearStyle,
+    this.yearForegroundColor,
+    this.yearBackgroundColor,
+    this.yearOverlayColor,
+    this.rangePickerElevation,
+    this.rangePickerShadowColor,
+    this.rangePickerSurfaceTintColor,
+    this.rangePickerShape,
+    this.rangePickerHeaderBackgroundColor,
+    this.rangePickerHeaderForegroundColor,
+    this.rangePickerHeaderHeadlineStyle,
+    this.rangePickerHeaderHelpStyle,
+    this.rangeSelectionBackgroundColor,
+    this.rangeSelectionOverlayColor,
+  });
+
+  /// The background color of a date picker dialog.
+  final Color? backgroundColor;
+
+  /// The elevation of a date picker dialog.
+  ///
+  /// {@macro flutter.material.material.elevation}
+  final double? elevation;
+
+  /// The color of the shadow painted below a date picker dialog.
+  ///
+  /// {@macro flutter.material.material.shadowColor}
+  final Color? shadowColor;
+
+  /// The color of the surface tint overlay applied to the [backgroundColor]
+  /// to indicate elevation.
+  ///
+  /// {@macro flutter.material.material.surfaceTintColor}
+  final Color? surfaceTintColor;
+
+  /// The overall shape of a date picker dialog.
+  ///
+  /// {@macro flutter.material.material.shape}
+  final ShapeBorder? shape;
+
+  /// The header's background fill color.
+  final Color? headerBackgroundColor;
+
+  /// The color used for text labels and icons in the header of a date
+  /// picker dialog.
+  ///
+  /// This is used instead of any colors provided by [headerHeadlineStyle]
+  /// or [headerHelpStyle].
+  final Color? headerForegroundColor;
+
+  /// The text style used for the headline text of the header.
+  ///
+  /// The headline text is usually the currently selected date in the header.
+  ///
+  /// The color of the [headerHeadlineStyle] is not used directly, as
+  /// [headerForegroundColor] is used instead.
+  final TextStyle? headerHeadlineStyle;
+
+  /// The text style used for the help text of the header.
+  ///
+  /// The help text (also referred to as "supporting text" in the Material
+  /// spec) is usually a prompt to the user at the top of the header
+  /// (i.e. 'Select date').
+  ///
+  /// The color of the [headerHelpStyle] is not used directly, as
+  /// [headerForegroundColor] is used instead.
+  ///
+  /// See also:
+  ///   [DatePickerDialog.helpText], which specifies the help text.
+  final TextStyle? headerHelpStyle;
+
+  /// The text style used for the row of week day labels at the top
+  /// of the date picker grid.
+  final TextStyle? weekdayStyle;
+
+  /// The text style used for each individual day label in the grid of the
+  /// date picker.
+  ///
+  /// The color of the [dayStyle] is not used directly, as
```

Comment:
Same comment here about it sounding deprecated.

Reply:
Addressed in https://github.com/flutter/flutter/pull/119033/commits/024dad5d17523874a83ad6e336b52a9f3f9ea026